### ISSUE TITLE
[BUG] Path prefix is not supported for remote reindex

### ISSUE THREAD

**Issue by Semenchuk1997** [opensearch-project/OpenSearch#1385]
I'm trying to reindex from the remote host using /_reindex API.
![image](https://user-images.githubusercontent.com/28215686/137937377-a6150fa9-babc-456c-ae16-de3c29ff27b6.png)

The host URL is of the form `[scheme]://[host]:[port](/pathPrefix)`. However, the reindex request fails with `Path prefix is not supported for remote reindex` reason.  

Seems like the issue was resolved in Elasticsearch 7.0.0 https://github.com/elastic/elasticsearch/issues/22913

**Comment by andrross** [comment]
I'll take a look at this

**Comment by kartg** [comment]
By 11/12, we're aiming to reproduce the issue and outline scope of change.

This shouldn't repro in Opensearch since we forked from Elasticsearch 7.10.2

**Comment by Semenchuk1997** [comment]
Here is my cluster props
![image](https://user-images.githubusercontent.com/28215686/139815436-d939234f-baba-4b5d-baec-d20cac90f06e.png)

**Comment by andrross** [comment]
OpenSearch forked from Elasticsearch 7.10.3, and the change to support pathPrefix is [included in OpenSearch 1.0](https://github.com/opensearch-project/OpenSearch/blob/996d33adb22ac6a523962166f3677f8dc6ac9c1f/server/src/main/java/org/opensearch/index/reindex/ReindexRequest.java#L475-L478) for remote reindexing operations. If I attempt to reproduce the error as shown I get:

```
{
  "error": {
    "root_cause": [
      {
        "type": "x_content_parse_exception",
        "reason": "[1:144] [reindex] failed to parse field [source]"
      }
    ],
    "type": "x_content_parse_exception",
    "reason": "[1:144] [reindex] failed to parse field [source]",
    "caused_by": {
      "type": "illegal_argument_exception",
      "reason": "Unsupported fields in [remote]: [external]"
    }
  },
  "status": 400
}
```

The `external` field does not seem to be a valid parameter for the [remote reindexing request](https://opensearch.org/docs/latest/opensearch/reindex-data/#reindex-from-a-remote-cluster). The valid fields for the `remote` parameter are [`host`, `username`, `password`, `socket_timeout`, and `connect_timeout`](https://opensearch.org/docs/1.0/opensearch/rest-api/document-apis/reindex/) Can you retry the request without the `external` field? (Though I can't explain why you're getting a different error message than me)

**Comment by Semenchuk1997** [comment]
Without `external` field the response the same. 

I saw the message as yours in the code. However, in the Dev Tools the message the same as in my screenshots above

**Comment by andrross** [comment]
Can you clarify how you're running your OpenSearch cluster? Is it hosted in AWS OpenSearch Service or are you running the instance yourself?

**Comment by Semenchuk1997** [comment]
Amazon OpenSearch Service

**Comment by andrross** [comment]
I've confirmed that path prefix support for remote reindex is implemented in OpenSearch 1.0. I'm not sure what is the causing the error that you're seeing, but at this point I think [AWS support](https://console.aws.amazon.com/support/home#/case/create?issueType=technical) is the right place to get assistance with Amazon OpenSearch Service.

**Comment by anasalkouz** [comment]
Closing the issue, please feel free to re-open the issue incase you still have any questions or concerns.
